MetaMask/metamask-extension
# Update Text to use TS version of Box

## Explanation
Currently the `Text` component still uses the JS version of the `Box` component and blocks TypeScript migration. This PR updates the `Text` component to use the TS version of the `Box` component. I've also added a few Before/After screenshots in the code comments to ensure no visual regression

* Fixes #19569

## Screenshots/Screencaps



### Before

Showing original `Text` component in storybook to show no visual regressions

https://github.com/MetaMask/metamask-extension/assets/8112138/c2eb2700-1c70-4c8b-a441-acc8388f1754

### After

Showing updated `Text` component in storybook using the TS version of the `Box` component. Showing no visual regressions and all stories and controls are working as expected. I have made some small updates to more easily find the  replace `Typography` with `Text` section and linked the issue

https://github.com/MetaMask/metamask-extension/assets/8112138/d9e916d6-0052-4329-b265-93169e0ebd12

## Manual Testing Steps

- I've added a few screenshots in the code comments to show no visual regressions between `develop` and the updates in this PR

To check storybook
- Go to the latest build of storybook in this PR 
- Search `Text` in the search panel
- Check the controls, stories and docs are working as expected.

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
